File: LeetCode/cpp/1053.cpp
class Solution {
public:
    vector<int> prevPermOpt1(vector<int> &A) {
        int n = A.size();
        for (int i = n - 2; i >= 0; --i) {
            int k = 0, num = 0;
            for (int j = i + 1; j < n; ++j) {
                if (num <= A[j] && A[j] < A[i]) {
                    num = A[j];
                    k = j;
                }
            }
            if (k != 0) {
                swap(A[i], A[k]);
                return A;
            }
        }
        return A;
    }
};

File: LeetCode/python3/1053.py
class Solution:
    def prevPermOpt1(self, A: List[int]) -> List[int]:
        n = len(A)
        for i in range(n - 2, -1, -1):
            k, num = 0, 0
            for j in range(i + 1, n):
                if A[j] < A[i] and num <= A[j]:
                    num = A[j]
                    k = j
            if k != 0:
                A[i], A[k] = A[k], A[i]
                return A
        return A
